lib/util/cmcd_manager.js

```diff
@@ -193,11 +194,37 @@ shaka.util.CmcdManager = class {
       if (stream) {
         if (isMedia) {
           data.bl = this.getBufferLength_(stream.type);
+          if (data.ot !== ObjectType.TIMED_TEXT) {
+            data.dl = this.getRemainingBufferLength_(stream.type);
+          }
         }
 
         if (stream.bandwidth) {
           data.br = stream.bandwidth / 1000;
         }
+
+        if (stream.segmentIndex && segment) {
+          const iterator =
+              stream.segmentIndex.getIteratorForTime(segment.endTime);
+          if (iterator) {
+            const nextSegment = iterator.next().value;
+            if (nextSegment && nextSegment != segment) {
+              if (!shaka.util.ArrayUtils.equal(
+                  segment.getUris(), nextSegment.getUris())) {
+                data.nor = nextSegment.getUris()[0];
```

Comment:
No :( we don’t have the relative url at this point

Reply:
We have a utility function in the Common Media Library that could be used here. It would need to be ported to a closure style util until we can figure out how to get ES6 NPM packages imported into the project.

https://github.com/streaming-video-technology-alliance/common-media-library/blob/main/lib/src/utils/urlToRelativePath.ts

Reply:
Thanks!